File: packages/shared/src/workflow.ts
/** Mirrors flowforge::domain::WorkflowStatus (engine/include/flowforge/domain/workflow.hpp). */
export type WorkflowStatus = "pending" | "running" | "succeeded" | "failed" | "cancelled";

export interface WorkflowStep {
  id: string;
  name: string;
  job_id: string;
  depends_on: string[];
}

export interface Workflow {
  id: string;
  name: string;
  status: WorkflowStatus;
  steps: WorkflowStep[];
  created_at: string;
  updated_at: string;
}

export interface ListWorkflowsResponse {
  workflows: Workflow[];
}

File: engine/src/domain/workflow.cpp
#include "flowforge/domain/workflow.hpp"

namespace flowforge::domain {

std::string_view to_string(WorkflowStatus status) noexcept {
  switch (status) {
    case WorkflowStatus::Pending:
      return "pending";
    case WorkflowStatus::Running:
      return "running";
    case WorkflowStatus::Succeeded:
      return "succeeded";
    case WorkflowStatus::Failed:
      return "failed";
    case WorkflowStatus::Cancelled:
      return "cancelled";
  }
  return "unknown";
}

}  // namespace flowforge::domain
